gnd/src/main/java/com/google/android/gnd/util/GeoJsonParser.java

```diff
@@ -0,0 +1,224 @@
+     *
+     * <p>Interior rings, which describe holes in the polygon, are ignored.
+     */
+    private static Optional<GeoJsonTile> fromJsonObject(JSONObject jsonObject) {
+      try {
+        JSONObject geometry = jsonObject.getJSONObject("geometry");
+        JSONArray sw = geometry.getJSONArray("coordinates").getJSONArray(0);
+        JSONArray ne = geometry.getJSONArray("coordinates").getJSONArray(2);
+        String id = jsonObject.getString("id");
+        String url = jsonObject.getJSONObject("properties").getString("title");
+
+        double south = sw.getDouble(0);
+        double west = sw.getDouble(1);
+        double north = ne.getDouble(0);
+        double east = ne.getDouble(1);
+
+        LatLng[] coords = {
+          new LatLng(south, west),
+          new LatLng(south, east),
+          new LatLng(north, west),
+          new LatLng(north, east)
+        };
+
+        Cartesian cartesian = Cartesian.fromString(id);
+
+        return Optional.of(new GeoJsonTile(coords, cartesian, url));
+      } catch (JSONException e) {
+        Log.e(TAG, "failed to parse geoJSONPolygon", e);
+      }
+      return Optional.empty();
+    }
+
+    /**
+     * Returns true iff any of {@link GeoJsonTile#coordinates} are contained within {@param bounds}.
+     *
+     * <p>This method assumes {@param bounds} is larger than the bounds described by {@link
+     * GeoJsonTile#coordinates} and does not account for cases in which the inverse is true.
+     */
+    private boolean intersects(LatLngBounds bounds) {
+      return stream(this.coordinates).anyMatch(bounds::contains);
+    }
+  }
+
+  /**
+   * Converts a JSONArray to an array of JSONObjects. Provided for compatibility with java8 streams.
+   * JSONArray itself only inherits from Object, and is not convertible to a stream.
+   */
+  private static JSONObject[] toArray(JSONArray arr) {
+    JSONObject[] result = new JSONObject[arr.length()];
+
+    for (int i = 0; i < arr.length(); i++) {
+      try {
+        JSONObject o = arr.getJSONObject(i);
+        result[i] = o;
+      } catch (JSONException e) {
+        Log.e(TAG, "couldn't parse json", e);
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * Returns the immutable list of tiles specified in {@param geojson} that intersect {@param
+   * bounds}.
+   */
+  public ImmutableList<Tile> intersectingTiles(LatLngBounds bounds, File geojson) {
+    try {
+      InputStream is = new FileInputStream(geojson);
+      BufferedReader buf = new BufferedReader(new InputStreamReader(is));
+      String line = buf.readLine();
+      StringBuilder sb = new StringBuilder();
+      while (line != null) {
+        sb.append(line).append('\n');
+        line = buf.readLine();
+      }
+
+      JSONObject geoJson = new JSONObject(sb.toString());
+      JSONArray features = geoJson.getJSONArray("features");
+
+      stream(toArray(features))
+          .filter(GeoJsonParser::hasGeometry)
+          .filter(GeoJsonParser::isPolygon)
```

Comment:
This is wrong -.- Have to fix this.

Reply:
isPolygon expects a geometry object, but here we return the entire feature object (and we need to maintain it down stream). 

I should move these checks into the GeoJsonTile.fromJson method anyway.